src/main/java/com/google/sps/data/Election.java

```diff
@@ -42,6 +44,8 @@
   // represents their Key names.
   public abstract Set<Long> getPropositions();
 
+  public abstract ArrayList<EmbeddedEntity> getPollingStations();
```

Comment:
Can we return an immutableset here? Store it as an arraylist in datastore, but when you get it out, put it into a set. And make it immutable to reduce bugs. 

See: go/ej3e-17

Reply:
Got it! By the way, the go link to specific pages in the book is super cool

Reply:
Do you mean change it to an immutable set in the Election class itself, or when retrieved through the PollingStationServlet?

Reply:
In the election class itself, and just transform it from an ArrayList when you instantiate this class. i.e. 

public abstract ImmutableSet<EmbeddedEntity> getPollingStations();

See gianluca's similar code here:
https://github.com/googleinterns/step190-2020/blob/master/src/main/java/com/google/sps/data/Contest.java#L81